### PR TITLE
Revise some unnecessary broadcasting

### PR DESCRIPTION
**Pull request by fehiepsi**
It seems to me (not 100% sure) that it is not necessary to apply broadcasting in mean/variance for the following distributions.

Some other non-important changes are:
+ Correct the order of distributions
+ replace `lax.full` by `np.full`

@neerajprad I notice that the order of `total_count` and `probs` has been swapped in discrete distributions. This order is different from pytorch/tensorflow/scipy/r. Just wonder if we should swap it back?

### CODE REVIEW

**Comment by neerajprad** [comment]
@neerajprad I notice that the order of total_count and probs has been swapped in discrete distributions. This order is different from pytorch/tensorflow/scipy/r. Just wonder if we should swap it back?

The reason for that is that `logits/probs` are not keyword arguments since they are now separated but `total_count=1` is a keyword argument which must be specified last. That said, I'm making a change to the distribution classes so that this separation is only on the backend, and the external wrapper takes in either logits/probs and delegates to the right implementation. I'll ensure then that this wrapper matches the interface in PyTorch.

**Comment by fehiepsi** [comment]
Thanks for explaining @neerajprad !
